docs/libs/ml/index.md

```diff
@@ -20,8 +20,100 @@ specific language governing permissions and limitations
 under the License.
 -->
 
+The Machine Learning (ML) library for Flink is a new effort to bring scalable ML tools to the Flink
+community. Our goal is is to design and implement a system that is scalable and can deal with
+problems of various sizes, whether your data size is measured in megabytes or terabytes and beyond.
+We call this library FlinkML.
+
+An important concern for developers of ML systems is the amount of glue code that developers are
+forced to write [1] in the process of implementing an end-to-end ML system. Our goal with FlinkML
+is to help developers keep glue code to a minimum. The Flink ecosystem provides a great setting to
+tackle this problem, with its scalable ETL capabilities that can be easily combined inside the same
+program with FlinkML, allowing the development of robust pipelines without the need to use yet
+another technology for data ingestion and data munging.
+
+Another goal for FlinkML is to make the library easy to use. To that end we will be providing
+detailed documentation along with examples for every part of the system. Our aim is that developers
+will be able to get started with writing their ML pipelines quickly, using familiar programming
+concepts and terminology.
+
+Contrary to other data-processing systems, Flink exploits in-memory data streaming, and natively
+executes iterative processing algorithms which are common in ML. We plan to exploit the streaming
+nature of Flink, and provide functionality designed specifically for data streams.
+
+FlinkML will allow data scientists to test their models locally and using subsets of data, and then
+use the same code to run their algorithms at a much larger scale in a cluster setting.
+
+We are inspired by other open source efforts to provide ML systems, in particular
+[scikit-learn](http://scikit-learn.org/) for cleanly specifying ML pipelines, and Spark’s
+[MLLib](https://spark.apache.org/mllib/) for providing ML algorithms that scale with problem and
+cluster sizes.
```

Comment:
The reason we (me and @ktzoumas) added this was to set out the goals of the library, what differentiates the system from other available currently, and point out some features.

The attribution to libraries that inspired the design of this library is also warranted.

Which section you think should be removed?